src/main/java/org/opensearch/geospatial/ip2geo/processor/Ip2GeoProcessor.java

```diff
@@ -153,8 +152,8 @@ protected void executeInternal(
         datasourceFacade.getDatasource(datasourceName, new ActionListener<>() {
             @Override
             public void onResponse(final Datasource datasource) {
-                if (datasource == null) {
-                    handler.accept(null, new IllegalStateException("datasource does not exist"));
+                if (datasource == null || DatasourceState.AVAILABLE.equals(datasource.getState()) == false) {
```

Comment:
could you please put this check into a method with a name that reflects condition that we're checking here, it's a bit hard to read? Method can be named something like "isDataSourceUnavailable"